plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergPartitionEvolution.java

```diff
@@ -278,4 +278,15 @@ public void testNestedFieldChangePartitionTransform()
         assertThat(monthPartitionedFiles).hasSize(2);
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @Test
+    public void testPartitionRename()
+    {
+        String tableName = "test_partition_rename_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['par']) AS SELECT 1 as v, 11 as par", 1);
+        assertQuerySucceeds("ALTER TABLE " + tableName + " RENAME COLUMN par to par_renamed");
+
+        assertQuery("SELECT partition.par_renamed FROM \"" + tableName + "$partitions\"", "VALUES 11");
+        assertUpdate("DROP TABLE " + tableName);
+    }
```

Comment:
Could you please add another test that has partitioning by transforms just in case? e.g. `bucket(par, 5)`

Reply:
good point. 
this will cause nested field to return field name only. 
Thinking how to rebuild the `column.field` ...

Reply:
Updated the logic with a new `partitionNameTransform` as single place source of truth on partition column naming. 
Now both nested identify and bucketing tranforms are handled as new names.